app/models/menu/menu.rb

```diff
@@ -0,0 +1,420 @@
+        id: "hud-reports-#{report_name.downcase.gsub(' ', '-')}",
+      )
+      menu.add_child(item)
+    end
+
+    menu
+  end
+
+  def warehouse_reports_menu
+    Menu::Item.new(
+      user: user,
+      visible: ->(user) { user.can_view_any_reports? },
+      path: warehouse_reports_path,
+      title: Translation.translate('Warehouse Reports'),
+      id: 'warehouse-reports',
+    )
+  end
+
+  def clients_menu
+    menu = Menu::Item.new(
+      user: user,
+      # visible: ->(user) { user.can_manage_all_agencies? || user.can_manage_agency? || GrdaWarehouse::Config.client_search_available? && user.can_access_some_client_search? || user.can_access_some_cohorts? },
+      title: Translation.translate('Clients'),
+      icon: 'icon-users2',
+      id: 'clients',
+      match_pattern_terminator: '.*',
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_manage_all_agencies? },
+        path: assigned_all_agencies_path,
+        title: Translation.translate('All Assigned Clients'),
+      ),
+    )
+    title = if user.can_view_aggregate_health? || user.can_view_patients_for_own_agency?
+      Translation.translate('My Agency\'s HMIS Clients')
+    else
+      Translation.translate('My Agency\'s Clients')
+    end
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_manage_agency? },
+        path: assigned_agencies_path,
+        title: title,
+      ),
+    )
+    title = if user.can_view_aggregate_health? || user.can_view_patients_for_own_agency?
+      Translation.translate('My HMIS Clients')
+    else
+      Translation.translate('My Clients')
+    end
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.user_clients.exists? },
+        path: assigned_clients_path,
+        title: title,
+      ),
+    )
+
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { GrdaWarehouse::Config.client_search_available? && user.can_access_some_client_search? },
+        path: clients_path,
+        title: Translation.translate('Client Search'),
+      ),
+    )
+
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_access_some_cohorts? },
+        path: cohorts_path,
+        title: Translation.translate('Cohorts'),
+      ),
+    )
+
+    menu
+  end
+
+  def health_menu
+    menu = Menu::Item.new(
+      user: user,
+      title: Translation.translate('Care Hub'),
+      icon: 'icon-heart-empty',
+      id: 'care-hub',
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { GrdaWarehouse::Config.get(:health_emergency_tracing).present? && user.can_edit_health_emergency_contact_tracing? },
+        path: health_he_search_path,
+        title: "#{GrdaWarehouse::Config.currrent_health_emergency_tracing_title} #{Translation.translate('Contact Tracing')}",
```

Comment:
Seems like we have a pervasive typo (extra "r" in "current'). Maybe good opportunity to fix?
```suggestion
        title: "#{GrdaWarehouse::Config.current_health_emergency_tracing_title} #{Translation.translate('Contact Tracing')}",
```

Reply:
Yeah, pervasive typo.  I hadn't fixed because I thought it required a DB migration, but I was wrong!